messaginginapp/src/main/java/io/customer/messaginginapp/ui/InlineInAppMessageView.kt

```diff
@@ -0,0 +1,89 @@
+package io.customer.messaginginapp.ui
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.view.Gravity
+import android.widget.ProgressBar
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
+import androidx.annotation.StyleRes
+import androidx.core.graphics.drawable.DrawableCompat
+import io.customer.messaginginapp.R
+
+/**
+ * InlineInAppMessageView is a custom view that displays an inline in-app message for given
+ * elementId. The view shows a progress indicator while the message is loading and hides it once the
+ * message is displayed. If there is no message to display, the view will hide itself and display
+ * automatically when a new message is available.
+ *
+ * Example usage in XML:
+ *
+ * <io.customer.messaginginapp.ui.InlineInAppMessageView
+ *     android:id="@+id/inline_in_app_message"
+ *     android:layout_width="match_parent"
+ *     android:layout_height="wrap_content"
+ *     app:elementId="banner-message"/>
+ */
+class InlineInAppMessageView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    @AttrRes defStyleAttr: Int = 0,
+    @StyleRes defStyleRes: Int = 0
+) : InlineInAppMessageBaseView(context, attrs, defStyleAttr, defStyleRes),
+    InlineInAppMessageViewListener {
+    private val progressIndicator: ProgressBar = ProgressBar(context)
+
+    init {
+        var progressColor = resolveThemeColor(android.R.attr.colorControlActivated, Color.GRAY)
+        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.InlineInAppMessageView)
+        try {
+            elementId = typedArray.getString(R.styleable.InlineInAppMessageView_elementId)
+            if (typedArray.hasValue(R.styleable.InlineInAppMessageView_progressTint)) {
+                progressColor = typedArray.getColor(
+                    R.styleable.InlineInAppMessageView_progressTint,
+                    progressColor
+                )
+            }
+        } finally {
+            typedArray.recycle()
+        }
+        setupProgressIndicator(progressColor)
+        viewListener = this
```

Comment:
```suggestion
 context.obtainStyledAttributes(attrs, R.styleable.InlineInAppMessageView).use { typedArray -> 
            elementId = typedArray.getString(R.styleable.InlineInAppMessageView_elementId)
            
            val defaultProgressColor = resolveThemeColor(android.R.attr.colorControlActivated, Color.GRAY)
            val progressColor = if (typedArray.hasValue(R.styleable.InlineInAppMessageView_progressTint)) {
                typedArray.getColor(R.styleable.InlineInAppMessageView_progressTint, defaultProgressColor)
            } else {
                defaultProgressColor
            }
            
            setProgressColor(progressColor)   
        }
```

Can we use `use` extension which automatically handles resource cleanup?

Reply:
I actually [used `use` earlier](https://github.com/customerio/customerio-android/pull/520/commits/2110db6df8c7beb1646315049ff496172feb9bff), but [lint kept failing](https://github.com/customerio/customerio-android/actions/runs/14595042785/job/40938814762) so reverted to a simpler approach.